packages/swingset-liveslots/test/test-cache.js

```diff
@@ -0,0 +1,120 @@
+import test from 'ava';
+import '@endo/init/debug.js';
+
+import { makeCache } from '../src/cache.js';
+
+test('cache', t => {
+  const backing = new Map();
+  backing.set('key0', 'value0');
+
+  const log = [];
+  const readBacking = key => {
+    log.push(['read', key]);
+    return backing.get(key);
+  };
+  const writeBacking = (key, value) => {
+    log.push(['write', key, value]);
+    backing.set(key, value);
+  };
+  const deleteBacking = key => {
+    log.push(['delete', key]);
+    backing.delete(key);
+  };
+
+  const c = makeCache(readBacking, writeBacking, deleteBacking);
+  c.insistClear();
+
+  // new reads pass through immediately to backing store
+  t.is(c.get('key0'), 'value0');
+  t.is(c.get('key2'), undefined);
+  t.deepEqual(log, [
+    ['read', 'key0'],
+    ['read', 'key2'],
+  ]);
+  log.length = 0;
+
+  // more reads within the same crank do not
+  t.is(c.get('key0'), 'value0');
+  t.deepEqual(log, []);
+
+  // writes update the cache but do not write through to backing store
+  c.set('key1', 'value1');
+  t.deepEqual(log, []);
+
+  // reads are served from the cache
+  t.is(c.get('key1'), 'value1');
+  t.deepEqual(log, []);
+
+  // new writes update the cache
+  c.set('key1', 'value2');
+  t.deepEqual(log, []);
+  t.is(c.get('key1'), 'value2');
+  t.deepEqual(log, []);
+
+  c.set('key3', 'value3');
+  c.set('key2', 'value2');
+  t.deepEqual(log, []);
+
+  c.delete('key2');
+
+  // flush writes/deletes everything dirty, in sorted order
+  c.flush();
+  t.deepEqual(log, [
+    // key0 is not dirty
+    ['write', 'key1', 'value2'],
+    ['delete', 'key2'],
+    ['write', 'key3', 'value3'],
+  ]);
+  log.length = 0;
+  c.insistClear();
+
+  // and now the cache is empty, so reads pass through
+  t.is(c.get('key3'), 'value3');
+  t.is(c.get('key2'), undefined);
+  t.is(c.get('key0'), 'value0');
+  t.is(c.get('key1'), 'value2');
+  t.deepEqual(log, [
+    ['read', 'key3'],
+    ['read', 'key2'],
+    ['read', 'key0'],
+    ['read', 'key1'],
+  ]);
+  log.length = 0;
+
+  // flush sees nothing dirty
+  c.flush();
+  t.deepEqual(log, []);
+  c.insistClear();
+
+  t.is(c.get('key0'), 'value0');
+  const m = s => ({ message: s });
+  t.throws(() => c.insistClear(), m(/cache still has stash/));
+  c.flush();
+  t.deepEqual(log, [['read', 'key0']]);
+  log.length = 0;
+  c.insistClear();
+
+  c.set('key1', 'value3');
+  t.throws(() => c.insistClear(), m(/cache still has dirtyKeys/));
+  c.flush();
+  t.deepEqual(log, [['write', 'key1', 'value3']]);
+  log.length = 0;
+  c.insistClear();
```

Comment:
```suggestion
  t.is(c.get('key0'), 'value0');
  const m = s => ({ message: s });
  t.throws(() => c.insistClear(), m(/^cache still has stash/));
  c.flush();
  t.deepEqual(log, [['read', 'key0']]);
  log.length = 0;
  c.insistClear();

  c.set('key1', 'value3');
  t.throws(() => c.insistClear(), m(/^cache still has dirtyKeys/));
  c.flush();
  t.deepEqual(log, [['write', 'key1', 'value3']]);
  log.length = 0;
  c.insistClear();
```
(or better without `m` IMO)